boards/shared/Test/Src/Test_ErrorTable.cpp

```diff
@@ -86,6 +88,52 @@ class SharedErrorTableTest : public testing::Test
                 error_table, DEFAULT_PDM_NON_CRITICAL_ERROR, true));
     }
 
+    template <typename T>
+    void TestRoutineForSetErrorsFromCanMsg(
+        enum Board                board,
+        std::vector<enum ErrorId> error_ids,
+        uint32_t                  can_stdid,
+        uint32_t                  can_dlc,
+        int (*pack_can_msg)(uint8_t *dst_p, const T *src_p, size_t size),
+        void (*get_boards_from_error_table)(
+            const struct ErrorTable *,
+            struct ErrorBoardList *),
+        void (*get_errors_from_error_table)(
+            struct ErrorTable *,
+            struct ErrorList *))
+    {
+        // Each struct member that has a non-zero value indicates an error that
+        // is set. For this test, we want every error to be set. Instead of
+        // setting each struct member manually, we're going to "cheat" using
+        // memset().
+        T can_data;
+        memset(&can_data, 1, sizeof(T));
+
+        // Prepare the CAN message containing the errors
+        can_msg.std_id = can_stdid;
+        can_msg.dlc    = can_dlc;
+        pack_can_msg(can_msg.data, &can_data, can_dlc);
+
+        // Update the error table using the given CAN message
+        Io_SharedErrorTable_SetErrorsFromCanMsg(error_table, &can_msg);
+
+        // Check that we can retrieve the correct board from the error table
+        get_boards_from_error_table(error_table, &board_list);
+        ASSERT_EQ(1, board_list.num_boards);
+        ASSERT_TRUE(App_SharedError_IsBoardInList(&board_list, board));
+
+        // Check that we can retrieve the correct errors from the error table
+        get_errors_from_error_table(error_table, &error_list);
+        ASSERT_EQ(error_ids.size(), error_list.num_errors);
+        for (auto &error_id : error_ids)
+        {
+            ASSERT_EQ(
+                EXIT_CODE_OK, App_SharedErrorTable_IsErrorSet(
+                                  error_table, error_id, &is_set));
+            ASSERT_TRUE(is_set);
+        }
```

Comment:
So this test tells us if every single error is set, then all the errors in the table will return true. Really what we want to check is that setting a _single_ error in the CAN message sets that error in the table, no?

Reply:
So I was thinking what we're really trying to check here. I believe the main goal is that we have a `SET_ERROR()` in `Io_Process{Board}{Critical|NonCritical}ErrorMsg` for every error. If we set every single error at once, we'll be able to catch any missing `SET_ERROR`. Writing one test per error just seemed a bit cumbersome and it'll be really easy to forget to write a test or two.

Reply:
Ok, I can live with this.